### PR TITLE
hi alex I fixed your shitty codes

### PR DESCRIPTION
**Pull request by SMJSGaming**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,42 +1 @@
...
-# Officialities and such
-
-GDAPI is not endorsed by RobTop. Geometry Dash is a registered trademark of RobTop Games AB.
-
-# [Features to be added](FeaturesToBeAdded.md)
-
-<div align="center">
-    <a href="https://discord.gg/cq2FKbb"><img src="https://canary.discordapp.com/api/guilds/467885469108142100/widget.png?style=banner2" alt="Join Discord Server"/></a>
-</div>
+just annoying a friend of mine
```
Comment (Altenhh) [inline README.md]: perfect

**Comment by Altenhh** [comment]
@AlFasGD your opinion?

File: GDAPI/GDAPI/Application/Database.cs [hunk 2]
```diff
@@ -47,23 +47,7 @@ public class Database
...
-        public event Action LevelDataSetCompleted;
-        /// <summary>Raised upon completion of the decryption of the gamesave string that was set.</summary>
-        public event Action GamesaveDecrypted;
-        /// <summary>Raised upon completion of the decryption of the level data string that was set.</summary>
-        public event Action LevelDataDecrypted;
-        /// <summary>Raised upon completion of retrieval of the folder names as specified in the newly set gamesave string.</summary>
-        public event Action FolderNamesRetrieved;
-        /// <summary>Raised upon completion of retrieval of the player name as specified in the newly set gamesave string.</summary>
-        public event Action PlayerNameRetrieved;
+        public Tas
```
Comment (AlFasGD) [inline GDAPI/GDAPI/Application/Database.cs]: Tas

File: GDAPI/GDAPI/Application/Database.cs [hunk 3]
```diff
@@ -94,22 +78,7 @@ public class Database
...
-
-        /// <summary>The decrypted form of the game manager.</summary>
-        public string DecryptedGamesave
-        {
-            get
-            {
-                if (decryptedGamesave == null)
-                {
-                    TryDecryptGamesave(File.ReadAllText(GameManagerPath), out decryptedGamesave);
+xt(GameManagerPath), out decryptedGamesave);
```
Comment (AlFasGD) [inline GDAPI/GDAPI/Application/Database.cs]: NO YOU SHOULD DEFINE XT FIRST

File: GDAPI/GDAPI/Application/Database.cs [hunk 4]
```diff
@@ -141,22 +110,7 @@ public string DecryptedLevelData
...
-        #region Constructors
-        /// <summary>Initializes a new instance of the <seealso cref="Database"/> class from the default database file paths.</summary>
-        public Database() : this(GDGameManager, GDLocalLevels) { }
-        /// <summary>Initializes a new instance of the <seealso cref="Database"/> class from custom database file paths.</summary>
-        /// <param name="gameManagerPath">The file path of the game manager file of the game.</param>
-        /// <param name="localLevelsPath">The file path of the local levels file of the game.</param>
-        public Database(string gameManagerPath, string localLevelsPath)
-        {
-            Task.Run(() => SetDecryptedGamesave(File.ReadAllText(GameManagerPath = gameManagerPath)));
+        /// e.ReadAllText(GameManagerPath = gameManagerPath)));
```
Comment (AlFasGD) [inline GDAPI/GDAPI/Application/Database.cs]: Bad documentation

File: GDAPI/GDAPI/Application/Database.cs [hunk 5]
```diff
@@ -214,18 +153,7 @@ public void CloneLevel(int index)
...
-            UpdateLevelData();
-        }
-        /// <summary>Creates a new level with the name "Unnamed {n}" and adds it to the start of the level list.</summary>
-        public Level CreateLevel() => CreateLevel($"Unnamed {GetNextUnnamedNumber()}", "", DefaultLevelString);
-        /// <summary>Creates a new level with a specified name and adds it to the start of the level list.</summary>
-        /// <param name="name">The name of the new level to create.</param>
-        public Level CreateLevel(string name) => CreateLevel(name, "", DefaultLevelString);
-        /// <summary>Creates a new level with a specified name and description and adds it to the start of the level list.</summary>
-        /// <param name="name">The name of the new level to create.</param>
+            fo create.</param>
```
Comment (AlFasGD) [inline GDAPI/GDAPI/Application/Database.cs]: Can't even spell to

File: GDAPI/GDAPI/Application/Database.cs [hunk 6]
```diff
@@ -534,101 +382,7 @@ private async Task GetCustomObjects()
...
-                }
-                while (nextDictionaryStartIndex > 2 && nextDictionaryStartIndex < songMetadataEndIndex);
-                int currentIndex = songMetadataStartIndex;
-                while ((currentIndex = decryptedGamesave.Find("<k>", currentIndex, songMetadataEndIndex) + 3) > 2)
-                {
-                    int startingIndex = decryptedGamesave.Find("</k><d>", currentIndex, songMetadataEndIndex) + 7;
-                    int endingIndex = currentIndex = decryptedGamesave.Find("</d>", startingIndex, songMetadataEndIndex);
-                    SongMetadataInformation.Add(SongMetadata.Parse(decryptedGamesave.Substring(startingIndex, endingIndex - startingIndex)));
-                }
+            int endInde
```
Comment (AlFasGD) [inline GDAPI/GDAPI/Application/Database.cs]: Indie company?

File: GDAPI/GDAPI/Application/Database.cs [hunk 7]
```diff
@@ -680,6 +434,6 @@ private static async Task PerformTaskWithInvocableEvent(Task task, Action invoca
             task.ContinueWith(_ => invocableEvent?.Invoke());
             await task;
         }
-        #endregion
+        #endregion he alex guess what, I got a 2.2 copy
```
Comment (AlFasGD) [inline GDAPI/GDAPI/Application/Database.cs]: OWO

File: GDAPI/GDAPI/Information/GeometryDash/ObjectLists.cs [hunk 8]
```diff
@@ -13,27 +13,4 @@ public static class ObjectLists
...
-        {
-            List<int> objects = new List<int>();
-            foreach (var f in typeof(ObjectLists).GetFields().Where(f => f.FieldType == typeof(int[]) && f.IsStatic))
-                objects.AddRange((int[])f.GetRawConstantValue());
-            objects.Sort();
-            return objects.ToArray();
-        }
-    }
-}
+        /// <summa
```
Comment (AlFasGD) [inline GDAPI/GDAPI/Information/GeometryDash/ObjectLists.cs]: Σούμα

File: README.md (hunk 1 above)
Comment (AlFasGD) [reply README.md]: :(

**Comment by SMJSGaming** [comment]
nah man, it doesn't have to work to be pure perfection

**Comment by Altenhh** [comment]
perfection > code working
yes

**Pull request merged**